app/src/test/java/org/mozilla/fenix/compose/cfr/CFRPopupFullScreenLayoutTest.kt

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.cfr
+
+import android.content.Context
+import android.graphics.PixelFormat
+import android.view.View
+import android.view.ViewManager
+import android.view.WindowManager
+import android.view.WindowManager.LayoutParams
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import androidx.lifecycle.findViewTreeLifecycleOwner
+import androidx.savedstate.ViewTreeSavedStateRegistryOwner
+import androidx.savedstate.findViewTreeSavedStateRegistryOwner
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.spyk
+import io.mockk.verify
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+
+@RunWith(FenixRobolectricTestRunner::class)
+class CFRPopupFullScreenLayoutTest {
+    @Test
+    fun `WHEN the popup is constructed THEN setup lifecycle owners`() {
+        val anchor = View(testContext).apply {
+            ViewTreeLifecycleOwner.set(this, mockk())
+            ViewTreeSavedStateRegistryOwner.set(this, mockk())
+        }
+
+        val popupView = spyk(CFRPopupFullScreenLayout("", anchor, mockk(), mockk()) {})
+
+        assertNotNull(popupView.findViewTreeLifecycleOwner())
+        assertEquals(
+            anchor.findViewTreeLifecycleOwner(),
+            popupView.findViewTreeLifecycleOwner()
+        )
+        assertNotNull(popupView.findViewTreeSavedStateRegistryOwner())
+        assertEquals(
+            assertNotNull(anchor.findViewTreeSavedStateRegistryOwner()),
+            assertNotNull(popupView.findViewTreeSavedStateRegistryOwner())
+        )
+    }
+
+    @Test
+    fun `WHEN the popup is dismissed THEN cleanup lifecycle owners and detach from window`() {
+        val context = spyk(testContext)
+        val anchor = View(context).apply {
+            ViewTreeLifecycleOwner.set(this, mockk())
+            ViewTreeSavedStateRegistryOwner.set(this, mockk())
+        }
+        val windowManager = spyk(context.getSystemService(Context.WINDOW_SERVICE) as WindowManager)
+        every { context.getSystemService(Context.WINDOW_SERVICE) } returns windowManager
+        val popupView = CFRPopupFullScreenLayout("", anchor, mockk(), mockk()) {}
+        popupView.show()
+
+        popupView.dismiss()
+
+        assertNull(popupView.findViewTreeLifecycleOwner())
```

Comment:
Does it make sense to check if it was not `null` before `dismiss` is called? Technically speaking, it could have been `null` all along :)

Reply:
My reasoning was that we have a test already just above for checking that upon instantiating the popupView the lifecycle owners will not be null.
I can add a check here also to be sure.